src/Microsoft.ML.Recommender/MatrixFactorizationTrainer.cs

```diff
@@ -110,40 +123,69 @@ public sealed class Options
             public string LabelColumnName;
 
             /// <summary>
-            /// Loss function minimized for finding factor matrices.  Two values are allowed, 0 or 12. The values 0 means traditional collaborative filtering
-            /// problem with squared loss. The value 12 triggers one-class matrix factorization for implicit-feedback recommendation problem.
+            /// Loss function minimized for finding factor matrices.
             /// </summary>
+            /// <remarks>
+            /// Two values are allowed, <see cref="LossFunctionType.SquareLossRegression"/> or <see cref="LossFunctionType.SquareLossOneClass"/>.
+            /// The <see cref="LossFunctionType.SquareLossRegression"/> means traditional collaborative filtering problem with squared loss.
+            /// The <see cref="LossFunctionType.SquareLossOneClass"/> triggers one-class matrix factorization for implicit-feedback recommendation problem.
+            /// </remarks>
             [Argument(ArgumentType.AtMostOnce, HelpText = "Loss function minimized for finding factor matrices.")]
             [TGUI(SuggestedSweeps = "0,12")]
             [TlcModule.SweepableDiscreteParam("LossFunction", new object[] { LossFunctionType.SquareLossRegression, LossFunctionType.SquareLossOneClass })]
-            public LossFunctionType LossFunction = LossFunctionType.SquareLossRegression;
+            public LossFunctionType LossFunction = Defaults.LossFunction;
 
+            /// <summary>
+            /// Regularization parameter.
+            /// </summary>
+            /// <remarks>
+            /// It's the weight of factor matrices' norms in the objective function minimized by matrix factorization's algorithm. A small value could cause over-fitting.
+            /// </remarks>
             [Argument(ArgumentType.AtMostOnce, HelpText = "Regularization parameter. " +
                 "It's the weight of factor matrices' norms in the objective function minimized by matrix factorization's algorithm. " +
```

Comment:
>norms [](start = 53, length = 5)

Frobenius Norm

Reply:
Are you trying to summon demon in my PR?

---
In reply to: [254911433](https://github.com/dotnet/machinelearning/pull/2451#discussion_r254911433) [](ancestors = 254911433)